routes/api/followRoute.js

```diff
@@ -0,0 +1,187 @@
+ *     tags:
+ *       - Delete following
+ *     parameters:
+ *       - in: query
+ *         name: offset
+ *         type: integer
+ *         required: false
+ *         enum:
+ *           - yes
+ *           - no
+ *       - name: no limit
+ *         type: integer
+ *         required: follower and followee Id
+ *         enum:
+ *           - yes
+ *           - no
+ *     responses:
+ *       200:
+ *         description: Successfully followed an author
+ *         schema:
+ *           type: strig
+ *           properties:
+ *             users:
+ *               type: string
+ *               description: flag a success message
+ *               items:
+ *                 type: string
+ *       404:
+ *         description: Invalid user
+ *         schema:
+ *           type: object
+ *           properties:
+ *             users:
+ *               type: string
+ *               description: no author returned
+ *               items:
+ *                 type: string
+ */
+router.delete(
+  '/profiles/:id/unfollow',
+  jwtValidator,
+  FollowUsersController.unfollow
+);
+
+/**
+ * @swagger
+ * /profiles/:id/followees:
+ *   get:
+ *     summary: Get all followers
+ *     description: Get the user followers
+ *     tags:
+ *       - my followers
+ *     parameters:
+ *       - in: query
+ *         name: offset
+ *         type: integer
+ *         required: false
+ *         enum:
+ *           - yes
+ *           - no
+ *       - name: limit
+ *         type: integer
+ *         required: false
+ *         enum:
+ *           - yes
+ *           - no
+ *     responses:
+ *       200:
+ *         description: Successfully fetch user followees
+ *         schema:
+ *           type: object
+ *           properties:
+ *             users:
+ *               type: string
+ *               description:
+ *               items:
+ *                 type: string
+ */
+router.get(
+  '/profiles/:id/followees',
+  jwtValidator,
+  FollowUsersController.followees
+);
+
+/**
+ * @swagger
+ * /profiles/:id/following:
+ *   get:
+ *     summary: Get all user following
+ *     description: Display all the users that the current user is following
+ *     tags:
+ *       - my followings
+ *     parameters:
+ *       - in: query
+ *         name: offset
+ *         type: integer
+ *         required: false
+ *         enum:
+ *           - yes
+ *           - no
+ *       - name: limit
+ *         type: integer
+ *         required: false
+ *         enum:
+ *           - yes
+ *           - no
+ *     responses:
+ *       200:
+ *         description: Successfully fetch user followers
+ *         schema:
+ *           type: object
+ *           properties:
+ *             users:
+ *               type: string
+ *               description:
+ *               items:
+ *                 type: string
+ */
+router.get(
+  '/profiles/:id/following',
+  jwtValidator,
+  FollowUsersController.following
+);
+
```

Comment:
The routes here should be in the users' route file. This is so because we are getting information related to a  user.

Reply:
Okay